packages/plugin-multi-tenant/src/types.ts

```diff
@@ -0,0 +1,82 @@
+import type { ArrayField, Field, RelationshipField, User } from 'payload'
+
+export type MultiTenantPluginConfig = {
+  collections: {
```

Comment:
It would be nice to allow (opt-in) generic typing to provide type-ahead and improve safety;

```ts
export type MultiTenantPluginConfig<TCollectionSlug extends string> = {
  collections: {
    [collectionSlug: TCollectionSlug]: { /* ... */ },
  },
  // ...
  tenantsSlug: TCollectionSlug
}
```

Which then could be used like that:

```ts
import type { Config } from "@/payload-types;

multiTenantPlugin<keyof Config['collections']>({})
```

Reply:
I implemented this, just in a bit of a different way using `key in CollectionSlug`, but I did use the generic to better type the `user` arg in the `userHasAccessToAllTenants` function. Thanks for the feedback!